### PR TITLE
Changes to Postable and Comment entities

### PR DESCRIPTION
**Pull request by izumidayo**
- created getReplies and getCreationDate methods on Postable
- added doc comments on class Comment and its methods
- overrode equals method for Comment
- added id attribute in Comment (implemented the same way as User)

### CODE REVIEW

**Review by sushimon — CHANGES_REQUESTED** [review]
check the comments i left. everything else lgtm!

File: src/main/java/entities/Postable.java [hunk 1]
```diff
@@ -1,15 +1,34 @@
 package entities;
 
 import java.util.List;
+import java.util.Date;
 
 abstract class Postable{
```
[the remark this answers is not in the document]
Comment (izumidayo) [reply src/main/java/entities/Postable.java]: done

File: src/main/java/entities/Comment.java [hunk 2]
```diff
@@ -7,30 +7,62 @@
 
 import java.security.PublicKey;
```
[the remark this answers is not in the document]
Comment (izumidayo) [reply src/main/java/entities/Comment.java]: removed!

**Review by sushimon — DISMISSED** [review]
lgtm!

File: src/main/java/entities/Comment.java [hunk 3]
```diff
@@ -5,32 +5,62 @@
...
+        super.creationDate = new Date();
+        Comment.numComments = numCommentsCreated;
+        Comment.numComments ++;
+        this.id = Comment.numComments;
     }
+
+    /**
+     * Returns the body (or text content) of this instance of Comment.
+     */
     public String getBody(){
```
Comment (reganq) [inline src/main/java/entities/Comment.java:39]: This method seems like something that the abstract class should already have implemented. We could change it to just be a method in super and then Comment and Post would already have these implemented

File: src/main/java/entities/Comment.java [hunk 4]
```diff
@@ -5,32 +5,62 @@
...
         return super.body;
     }
 
+    /**
+     * Returns the User that created this Comment.
+     */
     public User getUser(){
```
Comment (reganq) [inline src/main/java/entities/Comment.java:46]: Similar argument to line 24, in that this could also be a method in Postable.

**Review by reganq — CHANGES_REQUESTED** [review]
I think we should add getter/setter methods for the attributes of Postable, and then let Comment and Post inherit them. Other than that, everything looks good!

File: src/main/java/entities/Comment.java (hunk 3 above)
Comment (izumidayo) [reply src/main/java/entities/Comment.java]: moved!

**Review by BerryRahul — APPROVED** [review]

File: src/main/java/entities/Postable.java [hunk 5]
```diff
@@ -1,15 +1,53 @@
...
+
+    // the User who created the Postable
     User user;
+    // the main text content of the Postable
     String body;
-
+    // a list of Postable objects created as a reply to this Postable
     List<Postable> replies;
+    // the date this Postable was created
+    Date creationDate;
```
Comment (sushimon) [inline src/main/java/entities/Postable.java:17]: Maybe something to talk about with Johann. In the Post entity, he uses LocalDate to store the date of the deadline (different attribute but same idea) so maybe ask him just to be sure. 

Just as a note if you decide to change everything to LocalDate, `LocalDate.now()` returns the current date.

File: src/main/java/entities/Postable.java [hunk 6]
```diff
@@ -1,15 +1,53 @@
...
+
+    /**
+     * Returns the list of replies to a Postable object.
+     */
+    public List<Postable> getReplies(){return this.replies;}
+
+    /**
+     * Returns the date of which this Postable was created.
+     */
+    public Date getCreationDate(){return this.creationDate;}
```
Comment (sushimon) [inline src/main/java/entities/Postable.java:36]: Might need to change return type and docstring after discussing with Johann about whether to use `Date` vs `LocalDate`

File: src/main/java/entities/Postable.java [hunk 7]
```diff
@@ -1,15 +1,53 @@
 package entities;
 
 import java.util.List;
+import java.util.Date;
```
Comment (sushimon) [inline src/main/java/entities/Postable.java:4]: Might need to change if you decide to use `LocalDate` instead

File: src/main/java/entities/Comment.java [hunk 8]
```diff
@@ -5,32 +5,48 @@
...
+    public Comment(User commenter, String body, int numCommentsCreated){
         super.user = commenter;
         super.body = body;
         super.replies = new ArrayList<>();
-        creationDate = new Date();
-
-    }
-    public String getBody(){
-        return super.body;
+        super.creationDate = new Date();
```
Comment (sushimon) [inline src/main/java/entities/Comment.java:30]: Refer to other comments, similar idea.

**Review by sushimon — CHANGES_REQUESTED** [review]
Discuss with Johann about whether to use `Date` or `LocalDate` as it'd be weird to use two different types for two similar attributes imo

**Review by sushimon — APPROVED** [review]
Going to approve so we can get this merged, but something TODO is to discuss whether to use `Date` or `LocalDate`

**Pull request merged**